### PR TITLE
refactor: Convert ContentProviderTest to Kotlin

### PR DESCRIPTION
**Pull request by david-allison**

### CODE REVIEW

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 1]
```diff
@@ -309,261 +267,372 @@ public class ContentProviderTest extends InstrumentedTest {
...
 
-
     /**
      * Check that inserting a new model works as expected
      */
     @Test
-    public void testInsertAndUpdateModel() {
-        final ContentResolver cr = getContentResolver();
-        ContentValues cv = new ContentValues();
+    fun testInsertAndUpdateModel() {
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:551]: @KotlinCleanup("use apply when initializing ContentValues")

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 2]
```diff
@@ -573,178 +642,244 @@ public class ContentProviderTest extends InstrumentedTest {
...
-                FlashCardsContract.Note.CONTENT_URI,
-                FlashCardsContract.Model.CONTENT_URI,
-                FlashCardsContract.Deck.CONTENT_ALL_URI,
-                FlashCardsContract.Note.CONTENT_URI.buildUpon()
-                        .appendPath("1234")
-                        .appendPath("cards")
-                        .build(),
-        };
-        for (Uri uri : updateUris) {
+    fun testUnsupportedOperations() {
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:813]: @KotlinCleanup("use assertThrows")

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 3]
```diff
@@ -804,294 +963,380 @@ public class ContentProviderTest extends InstrumentedTest {
...
+        assertEquals("card is initial new", Consts.CARD_TYPE_NEW, card.queue)
+        val cr = contentResolver
+        val reviewInfoUri = FlashCardsContract.ReviewInfo.CONTENT_URI
+        val values = ContentValues()
+        val noteId = card.note().id
+        val cardOrd = card.ord
+        val earlyGraduatingEase =
+            if (schedVersion == 1) AbstractFlashcardViewer.EASE_3 else AbstractFlashcardViewer.EASE_4
+        val timeTaken: Long = 5000 // 5 seconds
+        values.put(FlashCardsContract.ReviewInfo.NOTE_ID, noteId)
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:1103]: @KotlinCleanup("move ContentValues instantiation here and use apply scope function")

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 4]
```diff
@@ -804,294 +963,380 @@ public class ContentProviderTest extends InstrumentedTest {
...
-
-        int updateCount = cr.update(reviewInfoUri, values, null, null);
-        assertEquals("Check if update returns 1", 1, updateCount);
+        val cr = contentResolver
+        val reviewInfoUri = FlashCardsContract.ReviewInfo.CONTENT_URI
+        val values = ContentValues()
+        val noteId = card.note().id
+        val cardOrd = card.ord
+        val bury = 1
+        values.put(FlashCardsContract.ReviewInfo.NOTE_ID, noteId)
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:1154]: @KotlinCleanup("move ContentValues instantiation here and use apply scope function")

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 5]
```diff
@@ -804,294 +963,380 @@ public class ContentProviderTest extends InstrumentedTest {
...
-
-        int updateCount = cr.update(reviewInfoUri, values, null, null);
-        assertEquals("Check if update returns 1", 1, updateCount);
+        val cr = contentResolver
+        val reviewInfoUri = FlashCardsContract.ReviewInfo.CONTENT_URI
+        val values = ContentValues()
+        val noteId = card.note().id
+        val cardOrd = card.ord
+        val suspend = 1
+        values.put(FlashCardsContract.ReviewInfo.NOTE_ID, noteId)
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:1204]: @KotlinCleanup("move ContentValues instantiation here and use apply scope function")

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 6]
```diff
@@ -804,294 +963,380 @@ public class ContentProviderTest extends InstrumentedTest {
...
-        values.put(FlashCardsContract.ReviewInfo.SUSPEND, suspend);
-
-        int updateCount = cr.update(reviewInfoUri, values, null, null);
-        assertEquals("Check if update returns 1", 1, updateCount);
+        val cr = contentResolver
+        val reviewInfoUri = FlashCardsContract.ReviewInfo.CONTENT_URI
+        val values = ContentValues()
+        val noteId = card.note().id
+        val cardOrd = card.ord
+        val suspend = 1
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:1203]: @KotlinCleanup("rename, while valid suspend is a kotlin soft keyword")

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 7]
```diff
@@ -16,291 +16,249 @@
...
-    @Parameterized.Parameters
-    public static java.util.Collection<Object[]> initParameters() {
-        // This does one run with schedVersion injected as 1, and one run as 2
-        return Arrays.asList(new Object[][] { { 1 }, { 2 } });
-    }
+@KotlinCleanup("IDE Lint")
+@KotlinCleanup("is -> equalTo")
+@KotlinCleanup("remove unnecessary .toLong() calls")
+@RunWith(Parameterized::class)
+class ContentProviderTest : InstrumentedTest() {
```
Comment (lukstbit) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:65]: @KotlinCleanup("try to replace try{query database...}finally{cursor.close()} with databaseQuery.use { cursor -> ...}")

**Review by lukstbit — CHANGES_REQUESTED** [review]

**Review by lukstbit — APPROVED** [review]

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 8]
```diff
@@ -16,291 +16,252 @@
...
-    }
+@KotlinCleanup("IDE Lint")
+@KotlinCleanup("is -> equalTo")
+@KotlinCleanup("remove unnecessary .toLong() calls")
+@KotlinCleanup("try to replace try{query database...}finally{cursor.close()} with databaseQuery.use { cursor -> ...}")
+@RunWith(Parameterized::class)
+class ContentProviderTest : InstrumentedTest() {
+    @JvmField
+    @Parameterized.Parameter
+    var schedVersion = 0
```
Comment (BrayanDSO) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:69]: lateinit

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 9]
```diff
@@ -16,291 +16,252 @@
...
-    private static final String[] TEST_MODEL_FIELDS = {"FRONTS","BACK"};
-    private static final String[] TEST_MODEL_CARDS = {"cArD1", "caRD2"};
-    private static final String[] TEST_MODEL_QFMT = {"{{FRONTS}}", "{{BACK}}"};
-    private static final String[] TEST_MODEL_AFMT = {"{{BACK}}", "{{FRONTS}}"};
-    private static final String[] TEST_NOTE_FIELDS = {"dis is za Fr0nt", "Te$t"};
-    private static final String TEST_MODEL_CSS = "styleeeee";
     // Whether tear down should be executed. I.e. if set up was not cancelled.
-    private boolean mTearDown;
+    private var mTearDown = false
+    private var mNumDecksBeforeTest = 0
```
Comment (BrayanDSO) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:79]: lateinit

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 10]
```diff
@@ -16,291 +16,252 @@
...
+        mCreatedNotes = ArrayList()
+        val col = col
 
         // We have parameterized the "schedVersion" variable, if we are on an emulator
         // (so it is safe) we will try to run with multiple scheduler versions
-        mTearDown = false;
-        if (InstrumentedTest.isEmulator()) {
-            col.changeSchedulerVer(schedVersion);
+        mTearDown = false
+        if (isEmulator()) {
```
Comment (BrayanDSO) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:108]: Keep `InstrumentedTest.`

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt (hunk 8 above)
Comment (david-allison) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:69]: On a `@Parameter§?

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt (hunk 10 above)
Comment (david-allison) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt]: This is now an IDE error

**Review by BrayanDSO — APPROVED** [review]

**Pull request merged**